rusoto/core/src/param.rs

```diff
@@ -9,16 +9,60 @@ pub type Params = BTreeMap<String, Option<String>>;
 
 /// Key:value pair for an service parameter.
 pub trait ServiceParams {
-    fn put(&mut self, key: &str, val: &str);
+    fn put<T: ToParam>(&mut self, key: &str, val: T);
     fn put_key(&mut self, key: &str);
 }
 
 impl ServiceParams for Params {
-    fn put(&mut self, key: &str, val: &str) {
-        self.insert(key.to_owned(), Some(val.to_owned()));
+    fn put<T: ToParam>(&mut self, key: &str, val: T) {
+        self.insert(key.to_owned(), Some(val.to_param()));
     }
 
     fn put_key(&mut self, key: &str) {
         self.insert(key.to_owned(), None);
     }
 }
+
+pub trait ToParam {
```

Comment:
I've now realized that this is actually not the proper way to encode these into the URL. Took some digging, but from what I can tell in the .NET SDK, these are actually supposed to each be individually added as a new parameter. e.g.

```rust
params.put("test", vec![0, 1, 2, 3]);
//=> GOOD: ?test=0&test=1&test=2&test=3
//=> BAD: ?test=0,1,2,3
```

I just tried generating `apigateway` and had another error here where it failed to add a map to the params. A map type in a query string is supposed to be looped over and each added individually.

I'll take a look at fixing these tomorrow.